src/coreclr/jit/compiler.cpp

```diff
@@ -9230,417 +9231,276 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
  *      cCVarSet,    dCVarSet       : Display a "converted" VARSET_TP: the varset is assumed to be tracked variable
  *                                    indices. These are converted to variable numbers and sorted. (Calls
  *                                    dumpConvertedVarSet()).
- *      cLoop,       dLoop          : Display the blocks of a loop, including the trees.
- *      cTreeFlags,  dTreeFlags     : Display tree flags
+ *      cLoop,       dLoop          : Display the blocks of a loop, including the trees, given a loop number (call
+ *                                    optPrintLoopInfo()).
+ *      cLoopPtr,    dLoopPtr       : Display the blocks of a loop, including the trees, given a LoopDsc* (call
+ *                                    optPrintLoopInfo()).
+ *      cLoops,      dLoops         : Display the loop table (call optPrintLoopTable()).
+ *      cTreeFlags,  dTreeFlags     : Display tree flags for a specified tree.
  *
  * The following don't require a Compiler* to work:
  *      dRegMask                    : Display a regMaskTP (call dspRegMask(mask)).
  *      dBlockList                  : Display a BasicBlockList*.
+ *
+ * The following find an object in the IR and return it, as well as setting a global variable with the value that can
+ * be used in the debugger (e.g., in the watch window, or as a way to get an address for data breakpoints).
+ *      dFindTreeInTree             : Find a tree in a given tree, specifying a tree id. Sets `dbTree`.
+ *      dFindTree                   : Find a tree in the IR, specifying a tree id. Sets `dbTree` and `dbTreeBlock`.
+ *      dFindStmt                   : Find a Statement in the IR, specifying a statement id. Sets `dbStmt`.
+ *      dFindBlock                  : Find a block in the IR, specifying a block number. Sets `dbBlock`.
+ *      dFindLoop                   : Find a loop in the loop table, specifying a loop number. Sets `dbLoop`.
  */
 
-void cBlock(Compiler* comp, BasicBlock* block)
+// Make the debug helpers available (under #ifdef DEBUG) even though they are unreferenced. When the Microsoft
+// linker is using /OPT:REF, it throws away unreferenced COMDATs (typically functions). Using "/INCLUDE:symbol"
+// forces the linker to keep these anyway.
+//
+// Mark them `export "C"` so the names are not C++ name mangled. This makes it easier to refer to them in the
+// `/INCLUDE` switch, and potentially makes them easier to find by a debugger.
+//
+// On x64, it appears the names are not decorated (which is actually unexpected). However, on x86, the names
+// are decorated for `__stdcall` with a leading `_` and trailing `@N` for `N` bytes of arguments. We could
+// make the functions all `__cdecl` to simplify, with only a leading `_`. For now, only do the "include"
+// thing for x64.
+
+#define DBGAPI extern "C"
+
+#if defined(HOST_AMD64) && defined(_MSC_VER)
```

Comment:
I assume it's disabled on x86 due to the weird mangling done there, but what about arm64?

Reply:
If this indeed is disabled on x86 for that reason I think you can fix it by switching these functions to stdcall (or cdecl, the opposite of the default we're using). In that case it's easier to keep the DBGAPI macro around.

Reply:
Yes, it's due to different mangling on x86. The mangling is due to the calling convention; we default to stdcall. We could have a separate set of pragmas for x86 mangling. It would be easier to support cdecl since it doesn't include the argument size (so isn't fragile to function signature changes). Then, using `DBGAPI` to specify cdecl could be used. I presume it would work for native arm64 because that and x64 only have a single calling convention so don't need to mangle.